android/src/main/java/com/appgoalz/reactnative/RNZoomBridgeModule.java

```diff
@@ -257,13 +257,13 @@ public void onMeetingStatusChanged(MeetingStatus meetingStatus, int errorCode, i
       return;
     }
 
-    if(meetingStatus == MeetingStatus.MEETING_STATUS_FAILED) {
+    if(meetingStatus == MeetingStatus.MEETING_STATUS_FAILED && errorCode != MeetingError.MEETING_ERROR_EXIT_WHEN_WAITING_HOST_START) {
```

Comment:
So when does the promise resolve when someone exits the waiting room?

Reply:
When someone exists, we get `MeetingStatus.MEETING_STATUS_IDLE` call and resolves successful.

Thats being handled on line number 266.